C-Plus-Plus/sort/merge_without_extra_space.cpp

```diff
@@ -0,0 +1,79 @@
+// C++ Program to Merge two sorted arrays without using extra space
+
+
+#include <bits/stdc++.h>
+using namespace std;
+
+//Function to merge two sorted arrays
+void merge(int arr1[], int arr2[], int n, int m) {
+	int i=n-1, j=0;
+	while(i>=0 && j<m){                         //Iterate through every element of arr1 and arr2 and compare elements of the arrays
+		if(arr2[j]<arr1[i]){
+	           swap(arr1[i], arr2[j]);
+	        }
+	           i--;
+	           j++;	    
+	}
+	sort(arr1, arr1+n);                         //sort the first array
+        sort(arr2, arr2+m);                         //sort the second array
+}
+
+
+
+int main() {
+        cout<<"Enter the size of the first array : ";
+        int n, m, i;
+        
+        cin >> n;
+        cout<<"Enter the size of the second array : ";
+        cin >> m;
+      
+        int arr1[n], arr2[m];
+        cout<<"Enter the elements of the first array : ";
+        for (i = 0; i < n; i++) {
+            cin >> arr1[i];
+        }
+        cout<<"Enter the elements of the second array : ";
+        for (i = 0; i < m; i++) {
+            cin >> arr2[i];
+        }
+        merge(arr1, arr2, n, m);                            //call the merge function on both the arrays
+        cout<<"Sorted Array : ";
```

Comment:
You need to add comments to your code. Make sure the comments you add are above the line of code and not alongside it.